CMakeLists.txt

```diff
@@ -65,11 +65,25 @@ option( DETRAY_BUILD_TEST_UTILS "Build the test utility library of Detray" OFF )
 option( DETRAY_BUILD_UNITTESTS "Build the unit tests of Detray" OFF )
 option( DETRAY_BUILD_INTEGRATIONTESTS "Build the integration tests of Detray"
    OFF )
+option( DETRAY_BUILD_ALL_TESTS "Build unit and integrations tests of Detray" OFF )
 option( DETRAY_BUILD_BENCHMARKS "Build the benchmark tests" OFF  )
+option( DETRAY_BUILD_CLI_TOOLS "Build the command line tools of Detray" OFF )
 option( DETRAY_BUILD_TUTORIALS "Build the tutorial executables of Detray" OFF )
 option( DETRAY_FAIL_ON_WARNINGS
    "Make the build fail on compiler/linker warnings" FALSE )
 
+# Need test functionality and svg display for command line tools
+if ( DETRAY_BUILD_CLI_TOOLS )
+   set( DETRAY_BUILD_TESTING ON )
+   set( DETRAY_SVG_DISPLAY ON )
```

Comment:
Can we just keep TEST_UTILS and ACTSSVG ON by default even though we turn off the tests by default

Reply:
I am not following, why do you want to do that? You can now do
```
cmake -B build --preset dev-fp32 .
```
followed by
```
cmake -B build --preset dev-fp32 -DDETRAY_BUILD_BENCHMARKS=ON .
```
followed by
```
cmake -B build --preset dev-fp32 -DDETRAY_BUILD_BENCHMARK=ON -DDETRAY_BUILD_UNITTESTS=ON  .
```
and it should not give you any errors. Is that not what you wanted?

Reply:
Yeah, This might do what I want to do partly :thinking: 
I don't know, what kind of decision making did you do when demoting `utils` while `io` `plugins` `detector` are still included by default?

Reply:
Core won't build without algebra plugins, so we need ```plugins```. As far as I can see, IO is needed for any experiment that wants to use a detray detector, since otherwise how would you build a detector in the first place? (I don't realistically believe any experiment will write their tracking geometry in C++ by using the ```detector_builder``` class directly). That being said, we can make the two csv readers/writers we have optional, of course. 
I am also fine with turning off the detector lib by default, it will only be needed for custom detector types and covfie fields. What I do not believe is, that any experiment is really interested in building the toy detector, etc. These are mostly meant for automatic testing and some for R&D, maybe small benchmarks. But for most R&D, the ODD (and extensions to it) should really be used imo.